woodwork/utils.py

```diff
@@ -465,3 +471,15 @@ def _infer_datetime_format(dates, n=100):
     except (TypeError, ValueError, IndexError, KeyError, NotImplementedError):
         mode_fmt = None
     return mode_fmt
+
+
+def replace_nan_empty_strings(df):
+    """Replaces empty string values with NaNs."""
+    df = df.fillna(value=np.nan)
+
+    for col, dtype in df.dtypes.items():
+        if str(dtype) not in ["boolean", "Int64", "Float64Dtype", "string"]:
```

Comment:
@tamargrey Looked into this a bit more. There does not seem to be a reason to skip `Int64` or `Float64` as far as I can tell. The `string` replacement works if we use `pd.NA` instead of `np.nan` as the replacement value. Still having trouble with the `boolean` dtype though - might be a pandas bug, but not sure?

Anyway, with the latest commit the replacement gets run on everything except `boolean`, but a `boolean` column will never contain string values anyway, so skipping it shouldn't be a big deal - just makes the code messier and more confusing.

Reply:
However the replacement on a `string` dtype fails on pandas 1.2.5, so in order to do that we will have to bump our min pandas requirement to 1.3.0 it seems.

@gsheni Any thoughts on doing that?

Reply:
> The `string` replacement works if we use `pd.NA` instead of `np.nan` as the replacement value

Nice! This will always be fine, right? Because `string` uses pd.NA anyway? Just would be worried about it ever causing issues for EvalML somehow.

> Still having trouble with the boolean dtype though - might be a pandas bug, but not sure

Feels like it would be a bug to me. It doesn't allow `.replace(val, np.nan)` with any `val` that's not True or False, while `bool` dtype will; it's just that it could never find anything to replace because the only values will be True or False. Even replacing pd.NA raises an error. 

Like you said, though, we're not losing anything by just skipping the replace call for `boolean` dtype (the case where a boolean column has a nan string would be if its dtype was `object`, I think anyway).

Reply:
FWIW, Featuretools and EvalML already both require `pandas>=1.3.0`.